src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -100,58 +196,293 @@ public String refreshOwned() {
-            Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository?public=false", token.getContent(), client);
-            
-            if (asString.isPresent()) {
-                RepoList repos = MAPPER.readValue(asString.get(), RepoList.class);
-                List<Container> containers = repos.getRepositories();
-                for (Container c : containers) {
-
-                    if (name == null ? (String) c.getName() == null : name.equals((String) c.getName())) {
-                        List<Container> list = containerDAO.findByNameAndNamespace(name, (String) c.getNamespace());
-
-                        if (list.isEmpty()) {
-                            long create = containerDAO.create(c);
-                            return containerDAO.findById(create);
-                        } else {
-                            System.out.println("Container already registered");
+    @ApiOperation(value = "Register a container", notes = "Register a container (public or private). Assumes that user is using quay.io and github", response = Container.class)
+    public Container registerContainer(@QueryParam("container_name") String name, @QueryParam("enduser_id") Long userId) throws IOException {
+        // User user = userDAO.findById(userId);
+        List<Token> tokens = tokenDAO.findByUserId(userId);
+
+        for (Token token : tokens) {
+            String tokenType = token.getTokenSource();
+            if (tokenType.equals(TokenType.QUAY_IO.toString())) {
+                Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository?last_modified=true&public=false",
+                        token.getContent(), client);
+
+                if (asString.isPresent()) {
+                    RepoList repos = MAPPER.readValue(asString.get(), RepoList.class);
+                    List<Container> containers = repos.getRepositories();
+                    for (Container c : containers) {
+
+                        if (name == null ? (String) c.getName() == null : name.equals((String) c.getName())) {
+                            String namespace = (String) c.getNamespace();
+                            List<Container> list = containerDAO.findByNameAndNamespaceAndRegistry(name, namespace, tokenType);
+
+                            if (list.isEmpty()) {
+                                String repo = namespace + "/" + name;
+                                Optional<String> asStringBuilds = ResourceUtilities.asString(TARGET_URL + "repository/" + repo + "/build/",
+                                        token.getContent(), client);
+                                String gitURL = "";
+
+                                if (asStringBuilds.isPresent()) {
+                                    String json = asStringBuilds.get();
+
+                                    // System.out.println(json);
+
+                                    Gson gson = new Gson();
```

Comment:
Please add a few comments here to explain what is occuring and what information you're trying to retrieve here.